content/docs/platform/integrations/email/writing-email-template.mdx

```diff
@@ -77,3 +107,44 @@ Show block can be used to conditionally show content based on a condition. Use t
 />
 
 In above example, if subscriber has `showTracking` field set to `true` in custom `data` attribute, then the `Track your order` button will be shown in the email sent to the subscriber. Similarly, payload variables can be used to conditionally show the content.
+
+## Code editor (custom HTML)
+
+Novu’s custom HTML for email lets you build fully custom email templates by writing raw HTML combined with full LiquidJS support for dynamic data binding, control flow, and personalization. This gives you full control over layout, structure, and rendering, beyond the capabilities of Novu's block editor.
+
+This feature supports advanced use cases such as:
+- Migrating existing HTML email templates.
+- Implementing full design systems.
+- Executing flow control logic directly inside email templates.
+- Complete layout and styling freedom without editor constraints.
+
+<Callout type='info'>Custom HTML is available for customers on the Pro tier and above.</Callout>
+
+### Authoring custom HTML emails
+
+You can create custom HTML emails using either the Novu dashboard (Code editor) or via API passthrough. Both methods provide full control over email rendering.
+
+#### Using the code editor in Novu dashboard
+
+The Code editor allows you to write raw HTML and embed LiquidJS expressions directly within the Novu dashboard.
+
+To use it:
+1. Open an email step in your workflow.
+2. Select the **Code editor** tab to switch to Custom HTML mode.
+3. Start writing your template using HTML and LiquidJS syntax.
+
+<Callout type='warn'>Once you switch to the Code editor, reverting the template back to the Block editor is not supported as some HTML constructs may not map cleanly to block definitions, so further edits must be made directly in HTML.</Callout>
+
+#### Using the API
+
+You can also submit fully pre-rendered HTML directly via the Novu API. This allows you to control template rendering server-side and submit the complete HTML payload for delivery.
+
+When using API passthrough:
+
+- Novu delivers the submitted HTML exactly as provided.
+- No additional wrappers, headers, or system-injected layout elements are added.
+- LiquidJS is not executed server-side; the HTML you submit must be fully rendered.
```

Comment:
`Novu delivers the submitted HTML exactly as provided` - During the sending phase Novu compiles the HTML + Liquid template in conjunction with payload and subscriber context information and delivers the fully rendered HTML email.

`No additional wrappers, headers, or system-injected layout elements are added` not true, we include the Novu branding footer by default.

`LiquidJS is not executed server-side...` as mentioned above, we compile the template during the sending phase, so it is executed on our end, but is saved in our DB as they send it (HTML+Liquid)